tox.ini

```diff
@@ -66,14 +66,15 @@ description = Run unit tests
 commands_pre =
     poetry install --only main,charm-libs,unit
 commands =
-    poetry run pytest --numprocesses=auto --cov=src --ignore={[vars]tests_path}/integration/ {posargs}
+    poetry run pytest --numprocesses=12 --cov=src --ignore={[vars]tests_path}/integration/ {posargs}
```

Comment:
Q: why this change?

Reply:
In this PR we're moving from 321 unittests to 777.

Execution is struggling on 2 workers.

I did quite some profiling around this, discussed with @carlcsaposs-canonical  (together with the change).
No anomalie was discovered (can provide details), certainly none relating to the changes of this PR.

Reply:
To be addressed within the scope of https://warthogs.atlassian.net/browse/DPE-2879